scopes/dependencies/dependency-resolver/dependency-installer.ts

```diff
@@ -103,8 +103,10 @@ export class DependencyInstaller {
       });
     }
 
-    // remove node modules dir for all components dirs, since it might contain left overs from previous install
-    await this.cleanCompsNodeModules(componentDirectoryMap);
+    if (!packageManagerOptions.rootComponents) {
+      // remove node modules dir for all components dirs, since it might contain left overs from previous install
+      await this.cleanCompsNodeModules(componentDirectoryMap);
```

Comment:
why if it's using root components we want to skip this cleaning? how is it related?

Reply:
I added a comment